app/uk/gov/hmrc/economiccrimelevyenrolment/navigation/EclReferencePageNavigator.scala

```diff
@@ -47,10 +55,24 @@ class EclReferencePageNavigator @Inject() (enrolmentStoreProxyConnector: Enrolme
       case Some(response) =>
         response.enrolments.find(_.identifiers.exists(_.value == eclReferenceNumber)) match {
           case Some(_) => routes.EclRegistrationDateController.onPageLoad()
-          case _       => routes.NotableErrorController.detailsDoNotMatch()
+          case _       =>
+            auditConnector.sendExtendedEvent(
+              ClaimEnrolmentDetailsMismatchAuditEvent(
+                internalId = internalId,
+                mismatchReason = ClaimEnrolmentDetailsMismatchReason.EclReferenceMismatch,
+                eclReference = eclReferenceNumber,
+                eclRegistrationDat = eclRegistrationDate
+              ).extendedDataEvent
+            )
+            routes.NotableErrorController.detailsDoNotMatch()
         }
       case _              => routes.NotableErrorController.detailsDoNotMatch()
     }
   }
 
+  private def extractEclRegistrationDate(userAnswers: UserAnswers): String =
+    userAnswers.eclRegistrationDate match {
+      case Some(date) => date.format(DateTimeFormatter.BASIC_ISO_DATE)
+      case _          => ""
+    }
```

Comment:
You do not need this - `eclRegistrationDate` is optional in the audit event as you may not have provided it yet

Reply:
Done :).